### ISSUE TITLE
Cannot run "make" 

### ISSUE THREAD

**Issue by pcesar22** [ZipCPU/autofpga#2]
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory --directory=sw
g++ -std=c++11 -g -Wall -I bus/ -c mapdhash.cpp -o obj-pc/mapdhash.o
mapdhash.cpp:292:43: error: cannot pass object of non-trivial type 'STRING' (aka 'basic_string<char,
      char_traits<char>, allocator<char> >') through variadic method; call will abort at runtime [-Wnon-pod-varargs]
                                gbl_msg.fatal("Could not parse %s\n", vl);
                                                                      ^
1 error generated.
make[1]: *** [obj-pc/mapdhash.o] Error 1
make: *** [sw] Error 2
```

I think this is related to C <-> C++ compatibility issues. I'm running on a macOS

https://stackoverflow.com/questions/23450300/error-cannot-pass-objects-of-non-trivially-copyable-type-stdstring-aka-clas/23450339

**Comment by pcesar22** [comment]
Commenting out that line results in another error:
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make --no-print-directory --directory=sw
bison --defines=expr.tab.h expr.ypp
expr.ypp:94.3-8: invalid directive: `%empty'
make[1]: *** [expr.tab.h] Error 1
make: *** [sw] Error 2
```

**Comment by ZipCPU** [comment]
[This patch](https://github.com/ZipCPU/autofpga/commit/8ec8302d75f606b96abaa0891e64a598a792c315) should fix your first issue.  Your second issue seems to be related to the version of bison you are running.  I'm currently running 3.0.4, with a copyright dating back to 2015.  Can you tell me which version you are running?

Dan

**Comment by pcesar22** [comment]
Thanks Dan. I was running: 

```
bison --version
bison (GNU Bison) 2.3
Written by Robert Corbett and Richard Stallman.

Copyright (C) 2006 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Yikes. This is default on macos. I ran `brew install bison` and followed the instructions to point $PATH to the new bison installation.

That got me past that part. Then, I got a similar issue as the first one:

```
bldcachable.cpp:179:5: error: cannot pass object of non-trivial type 'std::string' (aka 'basic_string<char, char_traits<char>,
      allocator<char> >') through variadic method; call will abort at runtime [-Wnon-pod-varargs]
                                (*fnamep));
                                ^
bldcachable.cpp:188:57: error: cannot pass object of non-trivial type 'STRING' (aka 'basic_string<char, char_traits<char>,
      allocator<char> >') through variadic method; call will abort at runtime [-Wnon-pod-varargs]
                        gbl_msg.error("Could not write cachable file: %s\n", fname);
```

But a similar change fixed those issues. Thanks!

**Comment by ZipCPU** [comment]
Should now be fixed in [5b57538](https://github.com/ZipCPU/autofpga/commit/5b57538081d60a04cd32aeba2436816062cb0141).

I really appreciate the feedback, though.  Please keep me posted, and let me know how this works out for you along the way.

Dan
